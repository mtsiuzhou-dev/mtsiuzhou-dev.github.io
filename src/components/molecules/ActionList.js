import { useDispatch } from 'react-redux';
import Button from '../atoms/Button';
import { clearCompletedTasks } from '../../store/TasksSlice';
import style from './ActionList.module.scss';

const ActionList = () => {
  const dispatch = useDispatch();
  return (
    <div className={style.actions}>
      <Button
        text="Clear completed"
        onClick={() => dispatch(clearCompletedTasks())}
      />
    </div>
  );
};
export default ActionList;
