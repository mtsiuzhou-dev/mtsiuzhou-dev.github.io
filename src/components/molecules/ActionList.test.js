import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import renderer from 'react-test-renderer';
import ActionList from './ActionList';
import TasksReducer from '../../store/TasksSlice';

describe('ActionList:', () => {
  const store = configureStore({
    reducer: {
      todo: TasksReducer,
    },
  });
  const actionList = (
    <Provider store={store}>
      <ActionList />
    </Provider>
  );

  test('renders correctly', () => {
    const actionListJson = renderer.create(actionList).toJSON();
    expect(actionListJson).toMatchSnapshot();
  });
});
