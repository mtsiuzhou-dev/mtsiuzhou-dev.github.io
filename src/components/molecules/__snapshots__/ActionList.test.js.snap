// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionList: renders correctly 1`] = `
<div
  className="actions"
>
  <button
    aria-pressed={false}
    className="button"
    onClick={[Function]}
    type="button"
  >
    Clear completed
  </button>
</div>
`;
